LookupAnything/DataParser.cs

```diff
@@ -445,30 +445,26 @@ private Item CreateRecipeItem(string? inputID, string outputID, MachineRecipeOut
             {
                 if (inputID != null)
                 {
-                    switch (outputID)
+                    switch (ItemRegistry.GetDataOrErrorItem(outputID).ItemId)
                     {
-                        case "(O)342":
                         case "342":
                             obj.preserve.Value = SObject.PreserveType.Pickle;
-                            obj.preservedParentSheetIndex.Value = inputID;
```

Comment:
We can create flavored items using the game's new methods for it now, instead of doing it manually. For example:
```cs
return ItemRegistry
    .GetObjectTypeDefinition()
    .CreateFlavoredPickle(ItemRegistry.Create(inputId));
```

Reply:
Done, though doesn't short circuit return to continue allow MachineRecipeOutputData to continue overriding if it exists